get-dependency-list.js

```diff
@@ -13,18 +13,23 @@ function ignoreMissing(dependency, optional, peerDependenciesMeta) {
     || peerDependenciesMeta && dependency in peerDependenciesMeta && peerDependenciesMeta[dependency].optional;
 }
 
-module.exports = function(filename, serverless) {
+module.exports = function(filename, serverless, _cache) {
   const servicePath = serverless.config.servicePath;
-
-  const filePaths = new Set();
   const modulePaths = new Set();
-
+  const filePaths = new Set();
+  const cache = _cache || new Set();
   const modulesToProcess = [];
   const localFilesToProcess = [filename];
 
   function handle(name, basedir, optionalDependencies, peerDependenciesMeta) {
     const moduleName = requirePackageName(name.replace(/\\/, '/'));
 
+    if (cache.has(`${basedir}:${name}`)) {
```

Comment:
Any concern about relative or symlinked paths colliding? Would using the pathToModule below as the cache key maybe avoid some of that?

Reply:
Yeah can I add a symlink in the repo and cover that case?

I'm thinking I should add an option to opt-in to the caching too just to be safe.

Reply:
yeah, I would feel better rolling such a thing out as opt-in if only initially in case we find bugs in our own services that we can cover before risking breaking other people's deployments

Reply:
Ok yeah, and I'm happy to keep this on my own branch for a while and test it out with our team's services for a while.